playground/src/components/selectors/experiment_submissions_select.tsx

```diff
@@ -0,0 +1,351 @@
+    trainingSubmissions?.some(
+      (usedSubmission) => usedSubmission.id === submission.id
+    ) ||
+    evaluationSubmissions?.some(
+      (usedSubmission) => usedSubmission.id === submission.id
+    );
+
+  const excludedSubmissions =
+    data?.filter((submission) => !isSubmissionUsed(submission)) ?? [];
+
+  const takeFromSubmissions = (submissions: Submission[]) => {
+    if (sumbissionsSelectType === "random") {
+      return [...submissions]
+        .sort(() => 0.5 - Math.random())
+        .slice(0, moveSubmissionsNumber);
+    } else {
+      return [...submissions].slice(0, moveSubmissionsNumber);
+    }
+  };
+
+  const moveSubmissions = (
+    from: "training" | "evaluation" | "excluded",
+    to: "training" | "evaluation" | "excluded"
+  ) => {
+    if (from === to || moveSubmissionsNumber <= 0) return;
+
+    if (from === "excluded") {
+      const moveSubmissions = takeFromSubmissions(excludedSubmissions);
+      if (to === "training") {
+        onChangeTrainingSubmissions([
+          ...(trainingSubmissions ?? []),
+          ...moveSubmissions,
+        ]);
+      } else if (to === "evaluation") {
+        onChangeEvaluationSubmissions([
+          ...(evaluationSubmissions ?? []),
+          ...moveSubmissions,
+        ]);
+      }
+    } else if (
+      from === "training" &&
+      trainingSubmissions
+    ) {
+      const moveSubmissions = takeFromSubmissions(trainingSubmissions);
+      const newTrainingSubmissions =
+        trainingSubmissions.filter(
+          (submission) => !moveSubmissions.some((s) => s.id === submission.id)
+        );
+      if (to === "excluded") {
+        onChangeTrainingSubmissions(newTrainingSubmissions);
+      } else if (to === "evaluation") {
+        onChangeTrainingSubmissions(newTrainingSubmissions);
+        onChangeEvaluationSubmissions([
+          ...(evaluationSubmissions ?? []),
+          ...moveSubmissions,
+        ]);
+      }
+    } else if (
+      from === "evaluation" &&
+      evaluationSubmissions
+    ) {
+      const moveSubmissions = takeFromSubmissions(evaluationSubmissions);
+      const newEvaluationSubmissions =
+        evaluationSubmissions.filter(
+          (submission) => !moveSubmissions.some((s) => s.id === submission.id)
+        );
+      if (to === "excluded") {
+        onChangeEvaluationSubmissions(newEvaluationSubmissions);
+      } else if (to === "training") {
+        onChangeEvaluationSubmissions(newEvaluationSubmissions);
+        onChangeTrainingSubmissions([
+          ...(trainingSubmissions ?? []),
+          ...moveSubmissions,
+        ]);
+      }
+    }
+  };
+
+  return (
+    <div className="flex flex-col">
+      <div className="text-lg font-bold">Submissions</div>
+      {!disabled && (
+        <div className="px-1">
+          <label className="flex items-center cursor-pointer">
+            <input
+              disabled={disabled}
+              type="checkbox"
+              checked={trainingSubmissions !== undefined}
```

Comment:
Could `trainingSubmissions` ever be `null`? In that case:
```suggestion
              checked={trainingSubmissions != undefined}
```

Reply:
Yeah, I think I'm overusing `!==` and `===` it cause me quite some headache. I changed it, just in case